agent/command/s3_get.go

```diff
@@ -177,7 +184,20 @@ func (c *s3get) Execute(ctx context.Context, comm client.Communicator, logger cl
 		return errors.Wrap(err, "validating expanded params")
 	}
 
-	c.assumedRoleARN = getAssumedRoleARN(conf, c.AwsSessionToken)
+	if expiration, found, err := getAssumedRoleExpiration(conf, c.AwsSessionToken); found {
+		c.assumedRoleARN = getAssumedRoleARN(conf, c.AwsSessionToken)
+		if err != nil {
+			c.existingCredentials = &aws.Credentials{
+				AccessKeyID:     c.AwsKey,
+				SecretAccessKey: c.AwsSecret,
+				SessionToken:    c.AwsSessionToken,
+				Expires:         expiration,
+				CanExpire:       true,
+			}
+		} else {
+			logger.Task().Errorf("Error parsing expiration time: '%s'. Continuing command by calling AssumeRole again.", err.Error())
```

Comment:
nit: might be better to use warning here since it doesn't result in a failure and has fallback behavior.